src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,164 @@
+import './CreateCourse.css';
+
+import Input from '../../common/Input/Input';
+import Textarea from '../../common/Textarea/Textarea';
+import Button from '../../common/Button/Button';
+import CreateAuthor from './components/CreateAuthor/CreateAuthor';
+import AuthorItem from './components/AuthorItem/AuthorItem';
+
+import {
+	buttonNames,
+	placeholders,
+	labels,
+	mockedAuthorsList,
+} from '../../constants';
+
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useNavigate } from 'react-router-dom';
+
+import { formatDuration, formatDate } from '../../helpers/durationFormatter';
+
+const CreateCourse = () => {
+	const navigate = useNavigate();
+
+	const [authors, setAuthors] = useState(mockedAuthorsList);
+	const [courseAuthors, setCourseAuthors] = useState([]);
+	const [newCourse, setNewCourse] = useState({
+		id: uuidv4(),
+		title: '',
+		description: '',
+		creationDate: formatDate(new Date()),
+		duration: 0,
+		authors: [],
+	});
+
+	const addNewAuthor = (author) => {
+		setAuthors((prevAuthors) => [...prevAuthors, author]);
+	};
+
+	const onAddCourseAuthor = (authorId) => {
+		const authorToAdd = authors.find((author) => author.id === authorId);
+		const updatedAuthors = authors.filter((author) => author.id !== authorId);
+
+		setAuthors(updatedAuthors);
+		setCourseAuthors([...courseAuthors, authorToAdd]);
+	};
+
+	const onDeleteCourseAuthor = (authorId) => {
+		const authorToDelete = courseAuthors.find(
+			(author) => author.id === authorId
+		);
+		const updatedCourseAuthors = courseAuthors.filter(
+			(author) => author.id !== authorId
+		);
+
+		setAuthors([...authors, authorToDelete]);
+		setCourseAuthors(updatedCourseAuthors);
+	};
+
+	const handleInputChange = (e) => {
+		setNewCourse({
+			...newCourse,
+			[e.target.id]:
+				e.target.id === 'duration' ? Number(e.target.value) : e.target.value,
+		});
+	};
+
+	const handleFormSubmit = async (e) => {
```

Comment:
For some reason, I'm not able to add more than 1 course. when  I add second course it replaces the previous one. 

https://github.com/VladimirSadovoi/ReactCourses/assets/157737324/581ce7ec-666a-46db-ba8f-e85e9b069477

Reply:
fixed